plugin/trino-hive/src/test/java/io/trino/plugin/hive/optimizer/TestHivePlans.java

```diff
@@ -268,6 +271,47 @@ public void testSubsumePartitionFilterNotConvertibleToTupleDomain()
                                                                         tableScan("table_unpartitioned", Map.of("R_STR_COL", "str_col", "R_INT_COL", "int_col"))))))))));
     }
 
+    @Test
+    public void testFilterDerivedFromTableProperties()
+    {
+        // Test that the filter is on build side table is derived from table properties
+        assertDistributedPlan(
+                "SELECT l.str_col, r.str_col FROM table_int_partitioned l JOIN table_unpartitioned r ON l.int_part = r.int_col",
+                noJoinReordering(),
+                output(
+                        exchange(REMOTE, GATHER,
+                                join(INNER, List.of(equiJoinClause("L_INT_PART", "R_INT_COL")),
+                                        exchange(REMOTE, REPARTITION,
+                                                project(
+                                                        filter("true", //dynamic filter
+                                                                tableScan("table_int_partitioned", Map.of("L_INT_PART", "int_part", "L_STR_COL", "str_col"))))),
+                                        exchange(LOCAL,
+                                                exchange(REMOTE, REPARTITION,
+                                                        project(
+                                                                filter("R_INT_COL IN (1, 2, 3, 4, 5)",
+                                                                        tableScan("table_unpartitioned", Map.of("R_STR_COL", "str_col", "R_INT_COL", "int_col"))))))))));
+    }
+
+    @Test
+    public void testFilterNotDerivedFromTablePropertiesForTooManyPartitions()
```

Comment:
You can add this test in the prep commit, like you did with the other test.
i understand the query fails before changes?